storage/client/client.go

```diff
@@ -802,9 +809,11 @@ func (c *StorageClient) DebondingDelegations(ctx context.Context, r *DebondingDe
 		if err != nil {
 			return nil, common.ErrStorageError
 		}
-		amount := new(big.Int).Mul(&shares, &escrowBalanceDebonding)
-		amount.Quo(amount, &escrowTotalSharesDebonding)
-		d.Amount = *amount
+		amount := new(big.Int).Mul(big.NewInt(shares.Int64()), big.NewInt(escrowBalanceDebonding.Int64()))
```

Comment:
Oy. If we cast our BigInt to int64, there's not much point in using arbitrary precision initially.
The wrapped types are hairy though, agreed :/.

Here's what I found:
According to https://go.dev/ref/spec#Struct_types, you can get the inner `big.Int` from a `BigInt` as `myBigInt.Int`.
So we can probably do math in `big.Int` space and then convert to `BigInt` at the end:
```go
amount := new(big.Int).Mul(&shares.Int, &escrowBalanceDebonding.Int)
amount.Quo(amount, &escrowTotalSharesDebonding.Int)
d.Amount = BigInt{*amount}
```

Ditto in line 740.